src/System.Diagnostics.TraceSource/tests/TraceSourceClassTests.cs

```diff
@@ -0,0 +1,101 @@
+        [InlineData(SourceLevels.Warning, TraceEventType.Warning, 1)]
+        [InlineData(SourceLevels.Warning, TraceEventType.Information, 0)]
+        [InlineData(SourceLevels.Information, TraceEventType.Information, 1)]
+        [InlineData(SourceLevels.Information, TraceEventType.Verbose, 0)]
+        [InlineData(SourceLevels.Verbose, TraceEventType.Verbose, 1)]
+        [InlineData(SourceLevels.All, TraceEventType.Critical, 1)]
+        [InlineData(SourceLevels.All, TraceEventType.Verbose, 1)]
+        // NOTE: tests to cover a TraceEventType value that is not in CoreFX (0x20 == TraceEventType.Start in 4.5)
+        [InlineData(SourceLevels.Verbose, (TraceEventType)0x20, 0)]
+        [InlineData(SourceLevels.All, (TraceEventType)0x20, 1)]
+        public void SwitchLevel(SourceLevels sourceLevel, TraceEventType messageLevel, int expected)
+        {
+            var trace = new TraceSource("TestTraceSource");
+            var listener = new TestTraceListener();
+            trace.Listeners.Add(listener);
+            trace.Switch.Level = sourceLevel;
+            trace.TraceEvent(messageLevel, 0);
+            Assert.Equal(expected, listener.GetCallCount(Method.TraceEvent));
+        }
+
+        [Fact]
+        public void TraceEvent()
+        {
+            var trace = new TraceSource("TestTraceSource", SourceLevels.All);
+            var listener = new TestTraceListener();
+            trace.Listeners.Add(listener);
+            trace.TraceEvent(TraceEventType.Verbose, 0);
+            Assert.Equal(1, listener.GetCallCount(Method.TraceEvent));
+        }
+
+        [Fact]
+        public void TraceData()
+        {
+            var trace = new TraceSource("TestTraceSource", SourceLevels.All);
+            var listener = new TestTraceListener();
+            trace.Listeners.Add(listener);
+            trace.TraceData(TraceEventType.Verbose, 0, null);
+            Assert.Equal(1, listener.GetCallCount(Method.TraceData));
+        }
+
+        [Fact]
+        public void TraceInformation()
+        {
+            var trace = new TraceSource("TestTraceSource", SourceLevels.All);
+            var listener = new TestTraceListener();
+            trace.Listeners.Add(listener);
+            trace.TraceInformation(null);
+            Assert.Equal(0, listener.GetCallCount(Method.TraceData));
+            Assert.Equal(0, listener.GetCallCount(Method.Write));
+            Assert.Equal(1, listener.GetCallCount(Method.TraceEvent));
+        }
+
+        [Fact]
+        public void Close()
+        {
+            var trace = new TraceSource("T1", SourceLevels.All);
+            trace.Listeners.Clear();
+
+            var listener = new TestTraceListener();
+            trace.Listeners.Add(listener);
+            trace.Close();
+            // NOTE: this assertion fails on .net 4.5
+            // where TraceSource.Close calls TraceListener.Close, not Dispose.
+            Assert.Equal(1, listener.GetCallCount(Method.Dispose));
```

Comment:
I would assume that Close and Dispose were deemed redundant; as I notice that Stream no longer has a Close method either. In the case of Stream, the base Dispose called Close, however for TraceListener, this wasn't true. I think this is why this unit test behavior is different on the two frameworks. The only TraceListener derived type in 4.5 that I see that looks like it would be impacted by the change in behavior is System.Diagnostics.Eventing.EventProviderTraceListener, which overrides Close, but not Dispose; I suspect that is likely a bug.